Refactor `MerchantEmailNotifications`

octaedro opened the pull request:
Fixes #8275

This PR refactors `MerchantEmailNotifications`.

- move src/Notes/MerchantEmailNotifications/NotificationEmail.php up into src/Notes (and keep it public)
- make /src/Notes/MerchantEmailNotificationss/MerchantEmailNotification.php internal
- rename `NotificationEmail` to `EmailNotification`

### Detailed test instructions:

- Check out this branch and create a `.zip` (by running `npm run test:zip`).
- Create a brand new site (using JN) and install and activate the `.zip`.
- Install the plugin [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/). This plugin will log every sent mail.
- Turn on the merchant email notifications by going to `WooCommerce` > `Settings` > `Emails` and enabling the checkbox next to `Enable email insights`.
- Install the plugin [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper).
- Go to `Tools` > `WCA Test Helper` > `Tools` and press the `Run` button located in the same row of `Update WC installation timestamp` and set a date to 4 days before. We want a 4-day-old store.
- In the same tab run the `wc_admin_daily` job. 
- Go to `Tools` > `WP Mail Log` and verify that an email with the subject `Add your first product` has been sent.
- Press `View` (under the Message column). The email should be visible now in a popup.

![screenshot-popular-ladybug jurassic ninja-2022 02 11-16_59_48](https://user-images.githubusercontent.com/1314156/153661795-49644d0e-a30c-4aca-845b-ccf77fb9bacc.png)


- Verify that the Unit tests are running correctly as well.

<!--
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note using the following CLI command `npm run changelogger -- add` and commit changes. --->

octaedro committed:
Refactor `MerchantEmailNotifications`

octaedro committed:
Rename `NotificationEmail`

octaedro committed:
Add changelog

louwie17 reviewed (commented):
@octaedro the code changes look fine to me, I just wasn't able to test it, it never send the email for me, but I was curious if that part depends on the [`notifications.json`](https://woocommerce.com/wp-json/wccom/inbox-notifications/1.0/notifications.json), which doesn't seem to contain the `merchant_notification` note?

I also wondered if we should make use of `class_alias` here also? I am leaning towards no, given this is a note, and it would seem odd for plugins to be depending on this, but I thought I mention it.

octaedro commented:
Hey @louwie17, 

> it never send the email for me, but I was curious if that part depends on the [notifications.json](https://woocommerce.com/wp-json/wccom/inbox-notifications/1.0/notifications.json), which doesn't seem to contain the merchant_notification note?

Yes, it doesn't contain that note. AFAIK, the only note with the type `email` that we have [is this](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Notes/AddFirstProduct.php). And to send the email it's necessary to:
1. turn on the email notes
2. have a store between 3 and 5 days old
3. not have products
4. not have orders

I tried this PR again and I didn't have any issues. Are you seeing any errors?

octaedro committed:
Add folder `Notes` to `Internal`

louwie17 reviewed (approved):
Aaah I got it working now, I thought it was anything greater then 4 day's, so I set my store to 5 day's old, which caused it not to work.
Tried it again and works as expected, this looks good to me, nice work @octaedro !

octaedro committed:
Moved folder and the namespace was renamed

octaedro commented:
Hey @louwie17,
Sorry for bugging you but I just rebased this PR, renamed the namespace, and moved the folder `src/Internal/Notes` to `src-interal/Admin`.
Could you take another look at this PR?

louwie17 reviewed (approved):
Thanks for the re-ping about the update, I just tested it again, and things still work well, also changes look good!
